File: 0001. Two Sum/Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = dict()
        for i in range(len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            else:
                d[target - nums[i]] = i

File: 0001. Two Sum/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        unordered_map<int, int> um;
        for (int i = 0; i < nums.size(); ++i) {
            if (um.find(nums[i]) != um.end())
                return {um[nums[i]], i};
            um[target - nums[i]] = i;
        }
        return {-1, -1};
    }
};